crates/matrix-sdk/src/client/builder.rs

```diff
@@ -439,20 +465,18 @@ impl ClientBuilder {
     }
 }
 
-fn homeserver_from_name(server_name: &ServerName) -> String {
-    #[cfg(not(test))]
-    return format!("https://{server_name}");
-
-    // Wiremock only knows how to test http endpoints:
-    // https://github.com/LukeMathWalker/wiremock-rs/issues/58
-    #[cfg(test)]
-    return format!("http://{server_name}");
+#[derive(Clone, Debug)]
+enum UrlScheme {
+    Http,
+    Https,
 }
 
 #[derive(Clone, Debug)]
 enum HomeserverConfig {
+    /// A precise URL, including the protocol.
     Url(String),
-    ServerName(OwnedServerName),
+    /// An host/port pair representing a server URL.
```

Comment:
```suggestion
    /// A host/port pair representing a server URL.
```

The h is not silent in English 😉